### DISCUSSION TITLE
About hiding *.lnk file path

### DISCUSSION THREAD

**Discussion by onesounds** [Flow-Launcher/Flow.Launcher#761] [Ideas]
This post is about a function that looks simple by hiding the path for uwp or lnk files. This function can be turned on/off by setting.

<img src="https://user-images.githubusercontent.com/6903107/137727590-5a859b57-8a13-4cc0-9947-2901cbabe3f3.png" width="400">

----

I thought it would be a problem to be solved by modifying only the win32 file and adding settings.
but when I looked at the code, it wasn't.
Since there are various plug-ins that provide file paths (program, Explorer, everything), all three plug-ins must be implemented to turn it on/off by setting it up. It is difficult to solve simply.

I think it would be nice to have a common converter that can be used in all three like a highlightconverter.
If proceeding, the following issues should also be included. (subtitle result path)
https://github.com/Flow-Launcher/Flow.Launcher/discussions/706

This problem is also linked to the path indication of the preview function.
https://github.com/Flow-Launcher/Flow.Launcher/pull/757

In addition, it is good to have an option to simply express the pre-registered route. 
For example, if the user registers c:\program files\dropbox as 'dropbox' in abbreviation, paths display such as c:\program files\dropbox\emoji\image\ will be changed to '·dropbox·emoji\image\'

All of these functions are for 'how to display a long file path to short'. If develop this function, should consider everything involved.

**Comment by jjw24** [comment]
> In addition, it is good to have an option to simply express the pre-registered route.
For example, if the user registers c:\program files\dropbox as 'dropbox' in abbreviation, paths display such as c:\program files\dropbox\emoji\image\ will be changed to '·dropbox·emoji\image'

Good idea for solving long path, but that probably means counting string length/manipulating string for every subtitle string, hmm, impact performance?

> uwp or lnk files

I don't use these types of programs enough, so why these two?

**Reply by onesounds** [reply]
Both files are often created by user installation. That is, there is a high probability that it is an app registered in the start menu or desktop. (The meaning of being registered in those two places means that it is often used. When I tested it, only those files were definitely applied.) These files are mainly used for execution and have the effect of removing the path of the file function to emphasize 'file running'. It's aesthetically good, too. This function was on listary 6, and I think this is a very good idea. If you actually use it, it's good because the executable is clearly distinguished. This option will certainly be useful because ordinary users use the flow to run the program rather than find the file.

**Reply by onesounds** [reply]
> Good idea for solving long path, but that probably means counting string lengthfor every subtitle string, hmm, impact performance?

Yeah. The importance is relatively low. If other functions are supported, the importance will be lower.

**Comment by onesounds** [comment]
https://github.com/Flow-Launcher/Flow.Launcher/pull/835 done.
